### ISSUE TITLE
Create rule S6850 (`jsx-a11y/heading-has-content`): Header elements should have accessible content

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#4383]
[jsx-a11y/heading-has-content](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/fffb05b38c8eee926ee758e9ceb9eae4e697fbdd/docs/rules/heading-has-content.md#jsx-a11yheading-has-content)
